File: 1/p1.py
import math


def fuel_for_mass(m):
    return int(math.floor(m / 3) - 2)


total_fuel = 0
with open('masses.txt') as masses:
    for l in masses.readlines():
        total_fuel += fuel_for_mass(int(l))

print 'total fuel', total_fuel

File: 1/p1.go
package main

import "io/ioutil"
import "fmt"
import "strconv"
import "strings"
import "math"


func fuelForMass (m int) int {
    return int(math.Floor(float64(m) / 3.0)) - 2
}

func main () {
    data, _ := ioutil.ReadFile("masses.txt")

    fuel := 0
    for _, s := range strings.Split(string(data), "\n") {
        i, err := strconv.Atoi(s)
        if err == nil {
            fuel += fuelForMass(i)
        }
    }

    fmt.Println("total fuel", fuel)
}
